iree/samples/dynamic_shapes/dynamic_shapes.ipynb

```diff
@@ -0,0 +1,555 @@
+          "text": [
+            "Zipping '/tmp/iree/colab_artifacts' to '/tmp/dynamic_shapes_colab_artifacts.zip' for download...\n",
+            "  adding: dynamic_shapes.mlir (deflated 90%)\n",
+            "Downloading the artifacts zip file...\n"
+          ],
+          "name": "stdout"
+        },
+        {
+          "output_type": "display_data",
+          "data": {
+            "application/javascript": [
+              "\n",
+              "    async function download(id, filename, size) {\n",
+              "      if (!google.colab.kernel.accessAllowed) {\n",
+              "        return;\n",
+              "      }\n",
+              "      const div = document.createElement('div');\n",
+              "      const label = document.createElement('label');\n",
+              "      label.textContent = `Downloading \"${filename}\": `;\n",
+              "      div.appendChild(label);\n",
+              "      const progress = document.createElement('progress');\n",
+              "      progress.max = size;\n",
+              "      div.appendChild(progress);\n",
+              "      document.body.appendChild(div);\n",
+              "\n",
+              "      const buffers = [];\n",
+              "      let downloaded = 0;\n",
+              "\n",
+              "      const channel = await google.colab.kernel.comms.open(id);\n",
+              "      // Send a message to notify the kernel that we're ready.\n",
+              "      channel.send({})\n",
+              "\n",
+              "      for await (const message of channel.messages) {\n",
+              "        // Send a message to notify the kernel that we're ready.\n",
+              "        channel.send({})\n",
+              "        if (message.buffers) {\n",
+              "          for (const buffer of message.buffers) {\n",
+              "            buffers.push(buffer);\n",
+              "            downloaded += buffer.byteLength;\n",
+              "            progress.value = downloaded;\n",
+              "          }\n",
+              "        }\n",
+              "      }\n",
+              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
+              "      const a = document.createElement('a');\n",
+              "      a.href = window.URL.createObjectURL(blob);\n",
+              "      a.download = filename;\n",
+              "      div.appendChild(a);\n",
+              "      a.click();\n",
+              "      div.remove();\n",
+              "    }\n",
+              "  "
+            ],
+            "text/plain": [
+              "<IPython.core.display.Javascript object>"
+            ]
+          },
+          "metadata": {
+            "tags": []
+          }
+        },
+        {
+          "output_type": "display_data",
+          "data": {
+            "application/javascript": [
+              "download(\"download_4cac3844-8109-4264-af93-11e6f5c12f9a\", \"dynamic_shapes_colab_artifacts.zip\", 1795)"
```

Comment:
same note about this being non-deterministic per invocation - is there a colab # or whatever command for not caching the output of a cell we could tag on this?

Reply:
There are ways to omit the output, yes. In this case I prefer for the output to remain so users browsing the notebook without running it can see what it does. There's already noise coming from all over when updating a notebook so I'm not sure how useful it would be to try to reduce that at this level :/

Reply:
this output in particular is the javascript download stuff - I see zero value in preserving that - no one is going to be able to do anything with it whether looking at the notebook rendered out or in text form

Reply:
Ugh, so `%%capture` ([docs](https://nbviewer.jupyter.org/github/ipython/ipython/blob/1.x/examples/notebooks/Cell%20Magics.ipynb#-Some-simple-cell-magics)) grabs stdout and stderr, which `google.colab.files.download()` relies on to function (it writes some html and javascript to the cell output that performs the download in the browser).

I could manually clear the output for just this cell (and remember to do that when updating the notebook).

I'm going to take a closer look at this as part of https://github.com/google/iree/issues/6165 anyway though. If we run the notebooks on CI as plain jupyter notebooks then this Colab file downloading might be a source of issues. I have ideas there, and they'll probably be different from what I can come up with just looking at this in isolation on this PR.

Reply:
I'm working on https://github.com/google/iree/issues/6165 now. 

As expected, the download is extra work to support:
* `ModuleNotFoundError` on `from google.colab import files`
* `python -m pip install google_colab` fails without Cython (and failing the install actually breaks other things: https://stackoverflow.com/q/64600508)

I'll just rip the downloads out and recommend users install from the file explorer in Colab...

(though I'd also prefer to do that _after_ merging this... editing Colab notebooks for individual changes takes an annoyingly long time)

Reply:
Okay, I manually removed the output for the download for now, and wrapped it with a try/catch so the notebook can run outside of Colab.